### PR TITLE
Refactor accessors

### PR DESCRIPTION
**Pull request by filiphr**
Split the PresenceChecks accessor from the current Accessor.
Introduce a ReadAccessor that would allow us to more easily implement certain things.
Remove MAP_GET and MAP_CONTAINS from the AccessorType and use the new refactored mechanism

I don't have an issue yet. However, if you (@Zegveld and @sjaakd) are OK with this change I'll create one when I do the merge.

The main idea behind this refactoring the the newly introduced `MAP_GET` and `MAP_CONTAINS` that leaked in places where they shouldn't have leaked.

### CODE REVIEW

**Review by Zegveld — COMMENTED** [review]
Didn't find anything that I object to, though I'm still not that familiar with the code.
Some minor remarks though, see the review.

File: processor/src/main/java/org/mapstruct/ap/internal/model/presence/SuffixPresenceCheck.java [hunk 1]
```diff
@@ -47,13 +47,12 @@ public boolean equals(Object o) {
         if ( o == null || getClass() != o.getClass() ) {
             return false;
         }
-        SourceReferenceMethodPresenceCheck that = (SourceReferenceMethodPresenceCheck) o;
-        return Objects.equals( sourceReference, that.sourceReference ) &&
-            Objects.equals( methodName, that.methodName );
+        SuffixPresenceCheck that = (SuffixPresenceCheck) o;
+        return Objects.equals( sourceReference, that.sourceReference );
```
[the remark this answers is not in the document]
Comment (filiphr) [reply processor/src/main/java/org/mapstruct/ap/internal/model/presence/SuffixPresenceCheck.java]: I don't think that we need the equals and hashcode, but you are right. Good spotting it. I've fixed it

File: processor/src/main/java/org/mapstruct/ap/internal/model/common/PresenceCheckAccessor.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.internal.model.common;
```
[the remark this answers is not in the document]
Comment (filiphr) [reply processor/src/main/java/org/mapstruct/ap/internal/model/common/PresenceCheckAccessor.java]: Good point. I had some other code in there initially that was creating package cycles. However, the `PresenceCheckAccessor` is now clean, so we can move it. Thanks for pointing it out

**Comment by filiphr** [comment]
Thanks for the review @Zegveld

**Pull request merged**